support/cas-server-support-pac4j-webflow/src/main/java/org/apereo/cas/web/flow/actions/DelegatedClientAuthenticationAction.java

```diff
@@ -134,6 +135,10 @@ protected Event doExecuteInternal(final RequestContext context) {
                     throw client.processLogout(callContext, clientCredential.get().getCredentials());
                 }
                 return finalizeDelegatedClientAuthentication(context, clientCredential.get());
+            } else if(StringUtils.isNotBlank(clientName)) {
+                // If there are no credentials at this point, it means that the
```

Comment:
Sorry, no this won't do. You need to either log this event with such messages, or you need to put this message into the exception itself because it's going to log an ERROR entry with an empty message (this might be the better option). Inline comments here don't help much (we can all tell what's happening here more or less)

Also, an alternative would be to just return:

```
return stopWebflow(e, context);
```

And that would be super clear, instead of logging, throwing, logging again and then returning.

Reply:
The problem with the ``return stopWebflow(e, context)`` approach is that the ``TechnicalException`` is logged and caught well before we get here.  Unless you mean
```
LOGGER.error("Client {} failed to validate credentials", clientName);
return stopWebflow(new AuthenticationException(), context);
```

Reply:
I see what you mean. Yes that would be fine, and for brevity, you could also do:

```java
return stopWebflow(new AuthenticationException("Client XYZ failed to validate credentials"), context);
```